src/components/HappinessTable.tsx

```diff
@@ -0,0 +1,177 @@
+import { useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { QueryKeys } from "../constants";
+import { useApi } from "../contexts/ApiProvider";
+import { Group } from "../data/models/Group";
+import { Happiness } from "../data/models/Happiness";
+import { floatToColor, formatDate, parseYYYYmmddFormat } from "../utils";
+import HappinessViewerModal from "./modals/HappinessViewerModal";
+/**
+ *
+ * @param startDate a string in the YYYY-MM-DD format representing the start of
+ * the happiness date range
+ * @param endDate a string in the YYYY-MM-DD format representing the end of
+ * the happiness date range (defaults to today if unspecified)
+ * @param groupId the associated group ID for the happiness table
+ * @returns
+ */
+export default function HappinessTable({
```

Comment:
is the table responsive in the horizontal direction? to view the entire table requires me to scroll left, which puts the sidebar out of view

Reply:
If you wrap the table in a parent of fixed width it scrolls properly. I also made the first columns sticky so as you scroll you can see whose happiness is whose.